src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable consistent-return */
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import classNames from 'classnames';
+import { TodoType } from '../../types/Todo';
+import { Todo } from '../Todo/Todo';
+import { useTodos } from '../../contexts/todosContext';
+
+type TodoListProps = {
+  todos: TodoType[];
+};
+
+export const TodoList = ({ todos }: TodoListProps) => {
+  const { tempTodo, processedTodos } = useTodos();
+
+  return (
+    <section className="todoapp__main">
+      <TransitionGroup>
+        {[...todos, tempTodo].map((todo) => {
+          if (!todo) {
```

Comment:
You should avoid returning undefined in a map function. Instead, you can use filter method to exclude undefined todos before mapping over the todos array.